src/hooks/useLedgerConnect.ts

```diff
@@ -1,103 +1,28 @@
-import AppEth from '@ledgerhq/hw-app-eth';
-import TransportBLE from '@ledgerhq/react-native-hw-transport-ble';
-import { forEach } from 'lodash';
-import { useCallback, useEffect, useState } from 'react';
-import useWallets from './useWallets';
-import { getHardwareKey, getHdPath, WalletLibraryType } from '@/model/wallet';
+import { useCallback, useEffect } from 'react';
 import { DebugContext } from '@/logger/debugContext';
-import { logger, RainbowError } from '@/logger';
-import * as i18n from '@/languages';
-
-enum LEDGER_ERROR_CODES {
-  OFF_OR_LOCKED = 'off_or_locked',
-  NO_ETH_APP = 'no_eth_app',
-  UNKNOWN = 'unknown',
-  DISCONNECTED = 'disconnected',
-}
-
-export const getLedgerErrorText = (errorCode: LEDGER_ERROR_CODES) => {
-  switch (errorCode) {
-    case LEDGER_ERROR_CODES.OFF_OR_LOCKED:
-      return i18n.t(i18n.l.hardware_wallets.errors.off_or_locked);
-    case LEDGER_ERROR_CODES.NO_ETH_APP:
-      return i18n.t(i18n.l.hardware_wallets.errors.no_eth_app);
-    default:
-      return i18n.t(i18n.l.hardware_wallets.errors.unknown);
-  }
-};
-
-export enum LEDGER_CONNECTION_STATUS {
-  LOADING = 'loading',
-  READY = 'ready',
-  ERROR = 'error',
-}
-
-/**
- * Parses ledger errors based on common issues
- */
-const ledgerErrorStateHandler = (error: Error) => {
-  if (error.message.includes('0x6511')) {
-    return LEDGER_ERROR_CODES.NO_ETH_APP;
-  }
-  if (
-    error.name.includes('BleError') ||
-    error.message.includes('0x6b0c') ||
-    error.message.includes('busy')
-  ) {
-    return LEDGER_ERROR_CODES.OFF_OR_LOCKED;
-  }
-
-  // used to logging any new errors so we can handle them properly, will likely remove pre-release
-  logger.warn('[LedgerConnect] - Unknown Error', { error });
-  forEach(Object.keys(error), key =>
-    // @ts-ignore
-    logger.debug('key: ', key, ' value: ', error[key])
-  );
-
-  return LEDGER_ERROR_CODES.UNKNOWN;
-};
+import { logger } from '@/logger';
+import { checkLedgerConnection, LEDGER_ERROR_CODES } from '@/utils/ledger';
 
 /**
  * React hook used for checking ledger connections and handling connnection error states
  */
-export function useLedgerStatusCheck({
-  address,
+export function useLedgerConnect({
```

Comment:
Nit: I don't see why we can't have most ledger stuff in the same file. Rn there's two separate hook files that don't really need to be hooks, and this one calls a util. They're all kinda utils in my mind 🤔